dm/pkg/checker/binlog.go

```diff
@@ -172,3 +170,73 @@ func (pc *MySQLBinlogRowImageChecker) Check(ctx context.Context) *Result {
 func (pc *MySQLBinlogRowImageChecker) Name() string {
 	return "mysql_binlog_row_image"
 }
+
+// BinlogDBChecker checks mysql/mariadb server ID.
+type BinlogDBChecker struct {
+	db      *sql.DB
+	dbinfo  *dbutil.DBConfig
+	schemas map[string]struct{}
+}
+
+// NewBinlogDBChecker returns a RealChecker.
+func NewBinlogDBChecker(db *sql.DB, dbinfo *dbutil.DBConfig, schemas map[string]struct{}) RealChecker {
+	return &BinlogDBChecker{db: db, dbinfo: dbinfo, schemas: schemas}
+}
+
+// Check implements the RealChecker interface.
+func (c *BinlogDBChecker) Check(ctx context.Context) *Result {
+	result := &Result{
+		Name:  c.Name(),
+		Desc:  "check whether dbs need replication is in binlog_do_db/binlog_ignore_db",
+		State: StateFailure,
+		Extra: fmt.Sprintf("address of db instance - %s:%d", c.dbinfo.Host, c.dbinfo.Port),
+	}
+
+	flavor, err := utils.GetFlavor(ctx, c.db)
+	if err != nil {
+		markCheckError(result, err)
+		return result
+	}
+	binlogDoDB, binlogIgnoreDB, err := utils.GetBinlogDB(ctx, c.db, flavor)
+	if err != nil {
+		markCheckError(result, err)
+		return result
+	}
+
+	binlogDoDB = strings.ReplaceAll(binlogDoDB, " ", "")
+	binlogIgnoreDB = strings.ReplaceAll(binlogIgnoreDB, " ", "")
+
+	binlogDoDBs := strings.Split(binlogDoDB, ",")
+	binlogIgnoreDBs := strings.Split(binlogIgnoreDB, ",")
+	// MySQL will check –binlog-do-db first, if there are any options,
+	// it will apply this one and ignore –binlog-ignore-db. If the
+	// –binlog-do-db is NOT set, then mysql will check –binlog-ignore-db.
+	// If both of them are empty, it will log changes for all DBs.
+	if len(binlogDoDB) != 0 {
+		for _, doDB := range binlogDoDBs {
+			delete(c.schemas, doDB)
```

Comment:
where c.chemas init?

Reply:
we also need to think about case-sensitive of this schemas

Reply:
> where c.chemas init?

https://github.com/pingcap/tiflow/blob/1e3dd155049417e3fd7bf9b0a0c7b08723b33791/dm/checker/checker.go#L214

> we also need to think about case-sensitive of this schemas

To be further confirmed. Because I think if schema name is case-sensitive, the `binlog_do_db/binlog_ignore_db` are too.

Reply:
After test, `binlog_db_db/binlog_ignore_db` is not affected by `lower_case_table_names`.

```sql
mysql> show master status;
+------------------+----------+----------------------+------------------+--------------------------------------------+
| File             | Position | Binlog_Do_DB         | Binlog_Ignore_DB | Executed_Gtid_Set                          |
+------------------+----------+----------------------+------------------+--------------------------------------------+
| mysql-bin.000004 |      194 | Do,TestCaseSensitive |                  | a0ec2f1a-d811-11eb-84a7-16210329b3ba:1-112 |
+------------------+----------+----------------------+------------------+--------------------------------------------+
1 row in set (0.00 sec)
```

I will add code about case-sensitive.